project/Build.scala

```diff
@@ -1153,6 +1164,14 @@ object Build {
   lazy val `scala3-bench-bootstrapped` = project.in(file("bench")).asDottyBench(Bootstrapped)
   lazy val `scala3-bench-run` = project.in(file("bench-run")).asDottyBench(Bootstrapped)
 
+  val testcasesOutputDir = taskKey[String]("Root directory where tests classses are generated")
+  val testcasesSourceRoot = taskKey[String]("Root directory where tests sources are generated")
+  val generateSelfDocumentation = inputKey[Unit]("Generate example documentation")
+  val generateScala3Documentation = taskKey[Unit]("Generate documentation for dotty lib")
+  val generateTestcasesDocumentation  = taskKey[Unit]("Generate documentation for testcases, usefull for debugging tests")
+  lazy val `scala3doc` = project.in(file("scala3doc")).asScala3doc
+  lazy val `scala3doc-testcases` = project.in(file("scala3doc-testcases")).asScala3docTestcases
```

Comment:
If they're just testcases, why are they in their own project in their own directory? Could they be in `scala3doc/test-resources` instead? (just like we have `compiler/test-resources`)

Reply:
We want to compile them since we are using generated .tasty files in our tests

Reply:
OK, so could `generateTestcasesDocumentation` be replaced by `scala3doc-testcases/doc` or is there a difference between the two?

Reply:
For now `scala3doc-testcases/doc` will not generate any documentation since we are not using (yet) scala3doc in any of our projects directly. This will involve working around cyclic depenencies and we can look at it after this PR is merged.